std/typecons.d

```diff
@@ -6778,6 +6909,23 @@ struct Typedef(T, T init = T.init, string cookie=null)
     }
 }
 
+///
+@safe unittest
```

Comment:
I'd remove the examples from the doc string

Reply:
There was actually a bit of value in them. So I kept them, but moved them to `unittest` blocks (see the second commit).

Reply:
(I had to rebase due to a merge conflict, so I squashed everything)